src/dataEntities/config.ts

```diff
@@ -131,6 +131,45 @@ export class ConfigManager {
             .filter(this.notEmpty)
             .reduce((a, b) => a.concat(b));
     }
+
+    /**
+     * Gets the current config settings, looking for both a configuration file and command line arguments.
+     * Command line arguments take precedence where present.
+     * Returns an error message of the command line args could not be parsed.
+     */
+    public getConfig() {
+        const fromCommandLine = this.fromCommandLineArgs(process.argv);
+        const returnConfig = Object.assign(config, fromCommandLine);
+        this.checkLogLevel(returnConfig.loglevel);
+        this.checkRateLimits(returnConfig);
+        return returnConfig;
+    }
+
+    private checkLogLevel(logLevel: string) {
+        const logLevelInfo = LogLevelInfo.tryParse(logLevel);
+        if (!logLevelInfo) throw new Error("Option 'loglevel' can only be one of the following: " + Object.values(LogLevel).join(", "));
+    }
+
+    private checkRateLimits(args: IArgConfig) {
+        if ((args.rateLimitUserWindowMs && !args.rateLimitUserMax) || (!args.rateLimitUserWindowMs && args.rateLimitUserMax)) {
+            throw new Error("Options 'rate-limit-user-windowms' and 'rate-limit-user-max' must be provided together.");
+        }
+
+        if ((args.rateLimitGlobalWindowMs && !args.rateLimitGlobalMax) || (!args.rateLimitGlobalWindowMs && args.rateLimitGlobalMax)) {
+            throw new Error("Options 'rate-limit-global-windowms' and 'rate-limit-global-max' must be provided together.");
+        }
+
+        if (args.maximumReorgLimit === 0) {
+            throw new Error("Option 'maximum-reorg-limit' cannot be 0.");
+        }
+    }
+}
+
+export class PisaConfigManager {
+    public static getConfig() {
+        const configManager = new ConfigManager(ConfigManager.PisaConfigProperties);
+        return configManager.getConfig()
+    }
```

Comment:
Not sure I get the design of these classes; if ConfigManager is generic while PisaConfigManager is tailored specifically for Pisa, then I think that `PisaConfigProperties` should be moved into `PisaConfigManager`.

Reply:
Agree, updated